src/commands/element.ts

```diff
@@ -15,28 +19,37 @@ export async function findElOrEls(
 ): Promise<any> {
    const driver = await getProxyDriver.bind(this)(strategy);
    let elementBody;
-   if (
-      !(driver instanceof JWProxy) &&
-      !(this.proxydriver instanceof AndroidUiautomator2Driver)
+   function constructFindElementPayload(
+      strategy: string,
+      selector: string,
+      proxyDriver: XCUITestDriver | AndroidUiautomator2Driver | Mac2Driver,
    ) {
-      elementBody = {
-         using: strategy,
-         value: selector,
-         context, //this needs be validated
-      };
-   } else {
-      elementBody = {
-         strategy,
-         selector: ['-flutter descendant', '-flutter ancestor'].includes(
-            strategy,
-         )
-            ? _.isString(selector)
-               ? JSON.parse(selector)
-               : selector
-            : selector,
-         context,
-      };
+      const isFlutterLocator =
+         strategy.startsWith('-flutter') || FLUTTER_LOCATORS.includes(strategy);
+
+      const parsedSelector =
+         ['-flutter descendant', '-flutter ancestor'].includes(strategy) &&
+         _.isString(selector)
+            ? JSON.parse(selector)
+            : selector; // Special case
+
+      if (
+         proxyDriver instanceof AndroidUiautomator2Driver ||
+         proxyDriver instanceof XCUITestDriver
+      ) {
```

Comment:
This condition still needs an update. 

```typescript
       // If user is looking for Native IOS/Mac locator
  if (!isFlutterLocator && proxyDriver instanceof XCUITestDriver || proxyDriver instanceof Mac2Driver) {
    return { using: strategy, value: parsedSelector, context };
  } else {
    return { strategy, selector: parsedSelector, context };
  }
```

Reply:
Along with the above changes, we need to update the unit tests to account for the new logic. 

All mocked driver implementations  in the tests as below

``` javascript
mockAppiumFlutterDriver.proxydriver = {
            constructor: { name: 'NotAndroidUiautomator2Driver' },
 };
```

should be replaced with `UiAutomatorDriver` , `XCUITestDriver` , and `Mac2Driver` to ensure proper coverage